packages/web-forms/src/components/FormHeader.vue

```diff
@@ -1,17 +1,86 @@
 <script setup lang="ts">
-import Card from 'primevue/card';
-defineProps<{title: string}>();
+import { type RootNode } from '@getodk/xforms-engine';
+import PrimeButton from 'primevue/button';
+import PrimeCard from 'primevue/card';
+import PrimeMenu from 'primevue/menu';
+import { ref } from 'vue';
+import FormLanguageDialog from './FormLanguageDialog.vue';
+import FormLanguageMenu from './FormLanguageMenu.vue';
+
+const props = defineProps<{form: RootNode}>();
+const languageDialogState = ref(false);
+const menu = ref<PrimeMenu>();
+const languages = props.form.languages.filter(language => !language.isSyntheticDefault);
+
+const print = () => window.print();
+
+const items = ref([
+	{
+		label: 'Print',
+		icon: 'icon-local_printshop',
+		command: print
+	}
+]);
+
+if(languages.length > 0){
+	items.value.unshift({
+		label: 'Change language',
+		icon: 'icon-language',
+		command: () => languageDialogState.value = true
+	})
+}
 </script>
+
 <template>
-	<Card class="form-title">
+	<!-- for desktop -->
+	<div class="hidden lg:flex justify-content-end flex-wrap gap-3">
+		<PrimeButton class="print-button" severity="secondary" rounded icon="icon-local_printshop" @click="print" />
+		<FormLanguageMenu :form="form" />
+	</div>
+	<PrimeCard class="form-title hidden lg:block">
 		<template #content>
-			<h1>{{ title }}</h1>
+			<!-- TODO/q: should the title be on the definition or definition.form be accessible instead of definition.bind.form -->
+			<h1>{{ form.definition.bind.form.title }}</h1>
 			<!-- last saved timestamp -->
 		</template>
-	</Card>
+	</PrimeCard>
+
+	<!-- for mobile and tablet -->
+	<div class="lg:hidden smaller-screens">
+		<div class="flex align-items-center">
+			<h1 class="flex-grow-1">
+				{{ form.definition.bind.form.title }}
+			</h1>
+
+			<!-- for tablet -->
+			<div class="odk-menu-bar hidden md:flex justify-content-end gap-3">
+				<PrimeButton class="print-button" severity="secondary" rounded icon="icon-local_printshop" @click="print" />
+				<FormLanguageMenu :form="form" />
+			</div>
+
+			<!-- for mobile -->
+			<PrimeButton icon="icon-menu" class="btn-menu md:hidden" text rounded aria-label="Menu" @click="menu?.toggle" />
+			<PrimeMenu id="overlay_menu" ref="menu" :model="items" :popup="true" />
+			<FormLanguageDialog v-model:state="languageDialogState" :form="form" />
+		</div>
+	</div>
 </template>
 
 <style scoped lang="scss">
+.p-button.p-button-icon-only.p-button-rounded {
+		height: 2.5rem;
+		width: 2.5rem;
+		min-width: 2.5rem;
+		font-size: 1.5rem;
+
+		&:hover{
+			background: var(--primary-100);
+		}
+		&:active, &:focus {
+			background: var(--primary-50);
+		}
```

Comment:
Unclear where to put this comment, but this is where it's most visible: the hamburger menu looks great in its default state, but when these background colors are applied it shows the background all the way to the right edge of the viewport. This is a minor nitpick, but it feels a little unpolished and maybe worth a little finesse.

Reply:
I had a discussion with Aly and she suggested to add equal spacing on the right side as there is on the left side, which is 10px.

Reply:
Now that you mention this, it caught my eye that quite a lot of measurements in the styles are in `px`. I don't think we should worry about this now, but I would like to give some thought into where relative units make sense for accessibility/scaling flexibility.

Reply:
Let's talk about this in Slack

Reply:
We talked about using `rem` instead of `px` in Slack and setting different font-size of root for different screen sizes. Tracking it in https://github.com/getodk/web-forms/issues/143